modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/view/podcast/PodcastFragment.kt

```diff
@@ -138,8 +138,18 @@ class PodcastFragment : BaseFragment(), Toolbar.OnMenuItemClickListener, Corouti
 
     override var statusBarColor: StatusBarColor = StatusBarColor.Custom(color = 0xFF1E1F1E.toInt(), isWhiteIcons = true)
 
-    private val onHeaderSummaryToggled: (expanded: Boolean) -> Unit = {
-        analyticsTracker.track(AnalyticsEvent.PODCAST_SCREEN_TOGGLE_SUMMARY, mapOf(IS_EXPANDED_KEY to it))
+    private val onHeaderSummaryToggled: (
+        podcastUuid: String,
+        expanded: Boolean,
+        userInitiated: Boolean,
+    ) -> Unit = { podcastUuid, expanded, userInitiated ->
+        if (userInitiated) {
+            analyticsTracker.track(
+                AnalyticsEvent.PODCAST_SCREEN_TOGGLE_SUMMARY,
+                mapOf(IS_EXPANDED_KEY to expanded)
+            )
+        }
+        if (BuildConfig.SHOW_RATINGS && expanded) viewModel.refreshPodcastRatings(podcastUuid)
```

Comment:
Ratings are refreshed whenever the podcast details view is expanded. We still need to decide on the refresh logic and whether ratings need to be displayed for subscribed podcasts.

Reply:
This will refresh the ratings each time the episode details are expanded, right? I'd be inclined to just refresh them once when the screen loads. Admittedly that would refresh the ratings in some cases where they aren't needed (if it's a subscribed podcast and the user never expands the details), but it also avoids any wait while the ratings are loaded if the user does expand the podcast details.

Reply:
> This will refresh the ratings each time the episode details are expanded, right? 

That's right.

>  I'd be inclined to just refresh them once when the screen loads. 

I was thinking of unnecessarily not querying ratings for subscribed podcasts until the user expands the details view. But I see your point. I'll take care of it in a future PR.